server/src/main/java/org/kaazing/gateway/server/context/resolve/DefaultServiceContext.java

```diff
@@ -282,9 +287,18 @@ public boolean equals(Object otherObject) {
 
     @Override
     public int hashCode() {
-        int result = serviceType != null ? serviceType.hashCode() : 0;
-        result = 31 * result + (accepts != null ? accepts.hashCode() : 0);
-        return result;
+        if (hashCode == -1) {
+            int result = serviceType != null ? serviceType.hashCode() : 0;
+            result = 31 * result + (accepts != null ? accepts.hashCode() : 0);
```

Comment:
Objects.hashCode(accepts) should correspond to (accepts != null ? accepts.hashCode() : 0)
Perhaps even better, why cannot we use Objects.hash(serviceType, accepts, serviceName)